app/Notifications/ChangeInFavoriteCity.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Notifications;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Notifications\Messages\MailMessage;
+use Illuminate\Notifications\Notification;
+
+class ChangeInFavoriteCity extends Notification
+{
+    use Queueable;
+
+    private string $city;
+    private string $provider;
+    private string $change;
+    private string $url;
+
+    public function __construct($city, $provider, $change)
+    {
+        $this->city = $city;
+        $this->provider = $provider;
+        $this->change = $change;
+        $this->url = env("APP_URL");
```

Comment:
Use variables from `config` files instead of `.env` file, u can cache these variables from config.
```suggestion
        $this->url = config("app.url");
```